packages/atomic/src/utils/slot-utils.ts

```diff
@@ -1,23 +1,31 @@
-export function getSlotFromHost(host: HTMLElement, slotName: string) {
+export function getNamedSlotFromHost(host: HTMLElement, slotName: string) {
   const children = Array.from(host.children);
-  const slots = children.filter((child) => {
-    if (slotName) {
-      return child.getAttribute('slot') === slotName;
-    }
+  const targetSlot = children.filter(
+    (child) => child.getAttribute('slot') === slotName
+  );
 
-    return child.getAttribute('slot') === '';
-  });
+  if (!targetSlot.length) {
+    return;
+  }
+
+  if (targetSlot.length > 1) {
+    console.warn(`Element should only have 1 slot named "${slotName}".`, host);
+  }
+
+  return targetSlot[0];
+}
+
+export function getDefaultSlotFromHost(host: HTMLElement) {
+  const children = Array.from(host.children);
+  const defaultSlot = children.filter((child) => !child.hasAttribute('slot'));
```

Comment:
it looks like prior to these changes, it was possible to have both a slot and a "default" slot by giving an element `slot=""`. Was this the case? That no longer looks possible here.

Reply:
```
<atomic-result-link>
  <h1>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>
```

Before:
<img width="269" alt="Screen Shot 2022-05-19 at 2 21 23 PM" src="https://user-images.githubusercontent.com/4923043/169372638-615a4d59-7532-47a9-bdfe-4ab49b7c523c.png">
<img width="269" alt="Screen Shot 2022-05-19 at 2 21 23 PM" src="https://user-images.githubusercontent.com/4923043/169372638-615a4d59-7532-47a9-bdfe-4ab49b7c523c.png">

After:
<img width="456" alt="Screen Shot 2022-05-19 at 2 20 33 PM" src="https://user-images.githubusercontent.com/4923043/169372666-f0255ce8-3119-4545-8f0a-a48fcd51199d.png">
<img width="456" alt="Screen Shot 2022-05-19 at 2 20 33 PM" src="https://user-images.githubusercontent.com/4923043/169372666-f0255ce8-3119-4545-8f0a-a48fcd51199d.png">

Reply:
```
<atomic-result-link>
  <h1>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
</atomic-result-link>

<atomic-result-link>
  <h1>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>

<atomic-result-link>
  <h1 part=''>hello</h1>
  <a part="attributes" target="_blank" title="HELLO"></a>
</atomic-result-link>
```

Before:
<img width="269" alt="Screen Shot 2022-05-19 at 2 21 23 PM" src="https://user-images.githubusercontent.com/4923043/169372638-615a4d59-7532-47a9-bdfe-4ab49b7c523c.png">
<img width="269" alt="Screen Shot 2022-05-19 at 2 21 23 PM" src="https://user-images.githubusercontent.com/4923043/169372638-615a4d59-7532-47a9-bdfe-4ab49b7c523c.png">

After:
<img width="456" alt="Screen Shot 2022-05-19 at 2 20 33 PM" src="https://user-images.githubusercontent.com/4923043/169372666-f0255ce8-3119-4545-8f0a-a48fcd51199d.png">
<img width="456" alt="Screen Shot 2022-05-19 at 2 20 33 PM" src="https://user-images.githubusercontent.com/4923043/169372666-f0255ce8-3119-4545-8f0a-a48fcd51199d.png">

Reply:
In the example you gave me, you're using `part`, did you test it using `part` or `slot`?

Reply:
ARGH always confuse those 2

Reply:
fixed!